components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -331,6 +331,18 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
         return OAuthUtil.buildConsumerAppDTO(app);
     }
 
+    private String[] filterOIDCAudience(OAuthConsumerAppDTO application) {
+
+        if (application.getAudiences() != null) {
```

Comment:
On second thoughts, wouldn't it be better to fail the requests if the client sends invalid or duplicate audiences?

This behaviour will be consistent with CORS origin validation logic as well.

Reply:
Decide to fail the request if empty or duplicate audience values were sent

Reply:
+1 to fail the request if the client send invalid audiences.